kafka-client/src/main/java/dev/responsive/kafka/store/ResponsiveRestoreListener.java

```
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2023 Responsive Computing, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.responsive.kafka.store;
+
+import dev.responsive.kafka.clients.ResponsiveRestoreConsumer;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.processor.StateRestoreListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Note: this is a global object which is shared across all StreamThreads and state stores in
+ * this Streams application.
+ */
+public class ResponsiveRestoreListener implements StateRestoreListener {
+  private static final Logger LOG = LoggerFactory.getLogger(ResponsiveRestoreListener.class);
+
+  private final Set<ResponsiveRestoreConsumer<?, ?>> restoreConsumers = new LinkedHashSet<>();
+  private StateRestoreListener userListener;
+
+  public void registerUserRestoreListener(final StateRestoreListener restoreListener) {
+    userListener = restoreListener;
```

should we throw an exception if `userListener` is already set?